src/wechaty_puppet/memory_card/memory_card.py

```diff
@@ -0,0 +1,332 @@
+        namespace = NAMESPACE_MULTIPLEX_SEPRATOR + NAMESPACE_MULTIPLEX_SEPRATOR.join(self.multiplexNameList)
+        return namespace
+
+    def _resolveKey(self, name: str) -> str:
+        if self.isMultiplex():
+            namespace = self._multiplexNamespace()
+            return NAMESPACE_KEY_SEPRATOR.join([namespace, name])
+        else:
+            return name
+
+    def isMultiplex(self) -> bool:
+        return len(self.multiplexNameList) > 0
+
+    def _multiplexPath(self) -> str:
+        return '/'.join(self.multiplexNameList)
+
+    def sub(self, name: str):
+        log.warning('MemoryCard', 'sub() DEPRECATED, use multiplex() instead')
+        return self.multiplex(name)
+
+    # FIXME Redeclared ?
+    def multiplex(self, name: str):
+        log.info('MemoryCard', 'multiplex(%s)' % name)
+
+        return self.multiplex(name)
+
+    async def destroy(self):
+        log.info('MemoryCard', 'destroy() storage: %s' % self.storage)
+
+        if self.isMultiplex():
+            raise Exception('can not destroy on a multiplexed memory')
+
+        await self.clear()
+
+        if self.storage:
+            await self.storage.destroy()
+            self.storage = None
+
+        self.payload = None
+
+    @property
+    def size(self) -> int:
+        log.info('MemoryCard', '<%s> size' % self._multiplexPath())
+        if not self.payload:
+            raise Exception('no payload, please call load() first.')
+        # FIXME
+        count: int = 0
+        if self.isMultiplex():
+            for item in filter(self._isMultiplexKey, self.payload.keys()):
+                if item:
+                    count += 1
+        else:
+            count = len(self.payload.keys())
+        return count
+
+    async def get(self, name: str):
+        log.info('MemoryCard', '<%s> get(%s)' % (self._multiplexPath(), name))
+
+        if not self.payload:
+            raise Exception('no payload, please call load() first.')
+
+        key = self._resolveKey(name)
+
+        return self.payload.get(key)
+
+    async def set(self, name, data):
+        log.info('MemoryCard', '<%s> set(%s, %s)', self._multiplexPath(), name, data)
+
+        if not self.payload:
+            raise Exception('no payload, please call load() first.')
+
+        key = self._resolveKey(name)
+        self.payload[key] = data
+
+    async def entries(self):
+        log.info('MemoryCard', '<%s> *entries()' % self._multiplexPath())
+        if not self.payload:
+            raise Exception('no payload, please call load() first.')
+
+        # while True:
+        #     try:
+        #         relativeKey = next(self.keys())
+        #         absoluteKey = self._resolveKey(relativeKey)
+        #         data = self.payload[absoluteKey]
+        #         pair = [relativeKey, data]
+        #         yield pair
+        #     except Exception as e:
+        #         pass
+        # pytype: disable=attribute-error
+        for relativeKey in self.keys():
```

Comment:
self.keys() is iterable

Reply:
All of the return type of the `keys`, `entries`, 'values'  should be `AsyncIterator`.